crates/client/src/commander.rs

```diff
@@ -282,104 +274,190 @@ impl Commander {
+                    let mut vec = c_shared_mutex.lock().unwrap();
+                    let mut vec_fuzzer = c_shared_mutex_fuzzer.lock().unwrap();
+
+                    vec.push(idl_program);
+                    vec_fuzzer.push((code, lib_path));
+
+                    c_mutex.store(false, std::sync::atomic::Ordering::SeqCst);
+                    Ok(())
+                } else {
+                    let error_text = String::from_utf8(output.stderr).unwrap();
+                    c_mutex.store(false, std::sync::atomic::Ordering::SeqCst);
+                    Err(Error::ReadProgramCodeFailed(error_text))
+                }
+            });
+
+            Self::expand_progress_bar(&package.name, &mutex);
+            cargo_thread.join().unwrap()?;
+        }
+        let idl_programs = shared_mutex.lock().unwrap().to_vec();
+        let codes_libs_pairs = shared_mutex_fuzzer.lock().unwrap().to_vec();
+
+        if idl_programs.is_empty() {
+            throw!(Error::NoProgramsFound);
+        } else {
+            (
+                Idl {
+                    programs: idl_programs,
+                },
+                codes_libs_pairs,
+            )
+        }
+    }
+    fn expand_package(package_name: &str) -> std::process::Output {
+        std::process::Command::new("cargo")
+            .arg("+nightly-2023-12-28")
+            .arg("rustc")
+            .args(["--package", package_name])
+            .arg("--profile=check")
+            .arg("--")
+            .arg("-Zunpretty=expanded")
+            .output()
+            .unwrap()
+    }
+
+    /// Returns `use` modules / statements
+    /// The goal of this method is to find all `use` statements defined by the user in the `.program_client`
+    /// crate. It solves the problem with regenerating the program client and removing imports defined by
+    /// the user.
     #[throws]
-    pub async fn get_programs_source_codes(&self) -> (Idl, Vec<(String, Utf8PathBuf)>) {
-        let program_idls_codes = self.program_packages().map(|package| async move {
-            let name = package.name;
-            let output = Command::new("cargo")
-                .arg("+nightly-2023-12-28")
-                .arg("rustc")
-                .args(["--package", &name])
-                .arg("--profile=check")
-                .arg("--")
-                .arg("-Zunpretty=expanded")
-                .output()
-                .await?;
+    pub async fn expand_program_client() -> Vec<syn::ItemUse> {
+        let shared_mutex = std::sync::Arc::new(std::sync::Mutex::new(String::new()));
+
+        let mutex = std::sync::Arc::new(std::sync::atomic::AtomicBool::new(true));
+        let c_mutex = std::sync::Arc::clone(&mutex);
+
+        let c_shared_mutex = std::sync::Arc::clone(&shared_mutex);
+
+        let cargo_thread = std::thread::spawn(move || -> Result<(), Error> {
+            let output = Self::expand_package("program_client");
```

Comment:
Yes I agree with this and will look at it. This expansion is already present within develop and within previous trdelnik versions so I did not want to increase the scope of the PR by removing/modifying it as you mentioned the PR is already pretty big. What about, if we eventually merge this PR, making this as next task for trdelnik instead of expanding this PR ?

Reply:
Yes, we can do...